### ISSUE TITLE
👥 Add contributor: phatblat

### ISSUE THREAD

**Issue by phatblat** [OAuthSwift/OAuthSwift#649]
Greetings! My team has started using this fabulous library in our inner-source auth library which will be used in 6 (and probably more) of our company's public iOS apps. We are using PingFederate as our OAuth server.

I have noticed that there are two new GitHub releases (2.1.1 & 2.1.2) that have been tagged but not yet released to CocoaPods trunk. Version 2.1.0 is the latest that shows up on CocoaPods.

```
-> OAuthSwift (2.1.0)
   Swift based OAuth library for iOS and macOS.
   pod 'OAuthSwift', '~> 2.1.0'
   - Homepage: https://github.com/OAuthSwift/OAuthSwift
   - Source:   https://github.com/OAuthSwift/OAuthSwift.git
   - Versions: 2.1.0, 2.0.0, 1.4.1, 1.4.0, 1.3.0, 1.2.2, 1.2.0, 1.1.2, 1.1.1, 1.1.0, 1.0.0, 0.6.0, 0.5.2, 0.5.1, 0.5.0, 0.4.8, 0.4.6, 0.4.5, 0.4.4, 0.4.3,
   0.3.7, 0.3.6, 0.3.5, 0.3.4, 0.3.3, 0.3.2, 0.3.1, 0.3.0, 0.2.1, 0.2.0, 0.1.9, 0.1.2, 0.1.1, 0.1.0 [cocoapods repo]
```

I would like to help out with publishing these and future releases to CocoaPods. I just need @phimage or @dongri to run the following command to add me as an owner of this pod.

```
pod trunk add-owner OAuthSwift ben@octop.ad
```

Note that all CocoaPods trunk session keys have been wiped due to a recent [trunk vulnerability](https://blog.cocoapods.org/CocoaPods-Trunk-RCE/), so you'll probably need to authenticate again with `pod trunk register`.

I have lots of experience publishing libraries to CocoaPods trunk and also publish our own 20+ internal iOS libraries to a private repo.

```
↪ pod trunk me
  - Name:     Ben Chatelain
  - Email:    ben@octop.ad
  - Since:    August 31st, 2015 19:47
  - Pods:
    - ObjectiveGit
    - libgit2
    - Nimble
    - Fetchable
    - BlueDot
    - Quick
    - RBQFetchedResultsController
    - ABFRealmMapView
    - RBQSafeRealmObject
    - SafeRealmObject
    - SwiftFetchedResultsController
    - RealmMapView
    - Outlets
    - Commandant
```

I can also help out with issue triage and PR review if you would be open to adding me as a contributor on this repo.

**Comment by mesopelagique** [comment]
Hi,

I have made cocoapod push (without any test, I do not like that 😭)
(change target tvos to 13 to make it pass)

I do not use anymore cocoapod so sorry for the late... 

Only @dongri could add you on github or cocoapod
I agree with that

**Comment by phimage** [comment]
Hi,

I have made cocoapod push (without any test, I do not like that 😭)
(change target tvos to 13 to make it pass)

I do not use anymore cocoapod so sorry for the late...

Only @dongri could add you on github or cocoapod
I agree with that

**Comment by dongri** [comment]
@phimage Thank you for your support!

@phatblat Hi, Ben
I have added you to cocoapod.
I've also added it to this repository

**Comment by mesopelagique** [comment]
for the 2.1.3 or 2.2.0

I have lost all my data, and so my test file with all my oauth service provider tokens , so without testing I am not confident

I try to figure the change log and see If no regression
https://github.com/OAuthSwift/OAuthSwift/compare/2.1.2...master

Checked
- add log system #586 by svoip (fixed compil issue  #612)
- readme cleaning
- reddit example #641 by mrnguuyen
- Mac Catalyst compile issue #605
- #599 cancel login on some callbackurl handler
(
#602 fix warning
 )

to check
- #623 signature plaintext ; TODO look at the RFC and check
- #596, ... change of refresh token ; execute the code one time, see it work (maybe compare with all code because code moved to client)
- ...

**Comment by phatblat** [comment]
Thanks! I look forward to working with you both and the users of this fabulous library!

I'm going to create a couple of issues to track the release of 2.1.1 and 2.1.2 to CocoaPods mostly as a place to note any hiccups or things I learn in the process. Meanwhile, I'll start reviewing the recent history to figure out whether the next release should be a patch or minor release.

I did notice that the public API was changed in 2.1.1 when `ASWebAuthenticationURLHandler` had the [`prefersEphemeralWebBrowserSession`](https://github.com/OAuthSwift/OAuthSwift/commit/3ae1a5d3335e0de90f22080e8637a2e600abd117#diff-422dc9b607338e5bda78df1aba80b93de81e6fae851adaf203f7cd304ff2e834) parameter added to the constructor. Technically, this is actually a breaking change since there is no default for the value, but would be a minor version if a default value were added so that the parameter didn't need to be added everywhere that `ASWebAuthenticationURLHandler` is constructed.

**Comment by phatblat** [comment]
Nevermind. Looks like 2.1.1 and 2.1.2 are live now.

**Comment by phimage** [comment]
I didn't care enough if semantic version rules has been broken

**Comment by phatblat** [comment]
I was wrong. There was a default value added in a subsequent commit. So, this was not a breaking API change in 2.1.1.

**Comment by phatblat** [comment]
It doesn't look like I have any permissions on this repo yet. I can't add labels or merge PRs. I wanted to merge #285 before releasing #620 since that's a good change, but I don't have permission.

**Comment by mesopelagique** [comment]
unfortunately I cannot do anything 😝    ( @dongri ?   maybe the role)
, except that I could merge or the moment
 #625 I suppose

before releasing something I want to check the #623 that I merge without really checking the code and result

**Comment by phatblat** [comment]
I'm a collaborator now that I found the invite that @dongri send me 😳

**Comment by phatblat** [comment]
Regarding #623, RFC-5849 section [3.4.4](https://datatracker.ietf.org/doc/html/rfc5849#section-3.4.4): PLAINTEXT

```
   The "PLAINTEXT" method does not employ a signature algorithm.  It
   MUST be used with a transport-layer mechanism such as TLS or SSL (or
   sent over a secure channel with equivalent protections).  It does not
   utilize the signature base string or the "oauth_timestamp" and
   "oauth_nonce" parameters.

   The "oauth_signature" protocol parameter is set to the concatenated
   value of:

   1.  The client shared-secret, after being encoded (Section 3.6).

   2.  An "&" character (ASCII code 38), which MUST be included even
       when either secret is empty.

   3.  The token shared-secret, after being encoded (Section 3.6).
```

The spec lines up with the implementation:

```swift
"\(consumerSecret)&\(oauthTokenSecret)"
```

However, the values used are *not* the `urlEncoded` values and thus could cause issues if one of those values contained unsupported characters.

**Comment by phimage** [comment]
thanks
So it's better than nothing

we could release or make fix with using maybe one of functions in https://github.com/OAuthSwift/OAuthSwift/blob/2384f60031ac824826fc367e6ed963b2c3eae797/Sources/String%2BOAuthSwift.swift

**Comment by phatblat** [comment]
Proposed a fix in #657

**Comment by phimage** [comment]
merged thanks
